esx_service/utils/vsan_info.py

```diff
@@ -56,11 +58,13 @@ def get_vsan_dockvols_path():
 
 def is_on_vsan(vmdk_path):
     """Returns True if path is on VSAN datastore, False otherwise"""
-    ds = get_vsan_datastore()
-    if not ds:
-        return False
+    global _vsan_ds
+    if not _vsan_ds:
+        _vsan_ds = get_vsan_datastore()
```

Comment:
When the service starts it calls init_datastoreCache(), can't the datastore type be be cached in that map as well in addition to the name, URL and dockvols path? This code path of is_vsan_datastore() seems to be extra when we have a DS cache already.

Reply:
I agree it's possible to improve&reuse existing datastore cache. Some minor concerns:
1. Adding a new entry to existing datastore cache might break existing code paths, so we need thorough regression test
2. The existing datastore cache is not very well encapsulated. The init_datastoreCache() is being called in multiple places. Ideally this should be avoided.

To reuse this cache, I will need to refactor existing implementation to improve the encapsulation and make it more robust. Otherwise I will have to call init_datastoreCache() in one more code path, which is really something I want to avoid.

Both are not critical concerns though. But it just means much more work to do. Considering that the current vsan_info.py is a module specifically for VSAN, I feel it's acceptable to maintain this minor cache in this module. So personally I don't have strong motivation to switch to the global datastore cache right now. Let me know what you think.